notebook/notebookapp.py

```diff
@@ -238,6 +238,20 @@ def init_handlers(self, settings):
         handlers.extend(load_handlers('services.nbconvert.handlers'))
         handlers.extend(load_handlers('services.kernelspecs.handlers'))
         handlers.extend(load_handlers('services.security.handlers'))
+        
+        # BEGIN HARDCODED WIDGETS HACK
+        try:
+            import ipywidgets
+            handlers.append(
+                (r"/nbextensions/widgets/(.*)", FileFindHandler, {
+                    'path': ipywidgets.find_static_assets(),
+                    'no_cache_paths': ['/'], # don't cache anything in nbextensions
```

Comment:
It probably is safe to cache our own widget js, right? It may be a bit annoying while developing them, but for users who install the package, it should be able to use the cache.